src/components/InputField/InputField.tsx

```diff
@@ -175,6 +186,19 @@ export const InputField: InputFieldType = forwardRef(
       fieldNote && styles['input-field--has-fieldNote'],
     );
 
+    const fieldLength = fieldText?.toString().length;
+
+    const textExceedsMaxLength =
+      maxLength !== undefined && fieldLength ? fieldLength > maxLength : false;
```

Comment:
Nit, couple ideas:

```ts
const fieldLength = fieldText?.toString().length ?? 0;
const textExceedsMaxLength = maxLength && fieldLength > maxLength;
```

or give `maxLength` a default and simplifyt the check even more

```ts
maxLength = Infinity

...

const textExceedsMaxLength = fieldLength > maxLength;
```

Reply:
Will apply a few things where possible. One thing I wanted to avoid were any props being added if the consumer didn't specify them. `maxLength` is also weird in that it changes how the base field behaves when it has a value, and it treats `Infinity` as invalid if it gets to `<textarea>`